### PR TITLE
refactor(STADTPULS-385): new supabase project

### PR DESCRIPTION
**Pull request by dnsos**
This PR refactors the whole codebase to use new Supabase instance, the updated schema, and the updated Stadtpuls API.

---

Also, includes the new sensor page, created in #122 

---

Note: There still remians a file `src/common/types/supabase_DEPRECATED.ts`. This is kept around for now as a reference. If we realize that the refactor went smooth, we can delete this.

This also applies to some files/components that won't be used in the new UI anymore (e.g. everything project concept-related or the now outdated `SensorsList`). Instead of deleting them already, they are kept until new versions are created.

### CODE REVIEW

File: src/lib/hooks/useUserData/index.ts [hunk 1]
```diff
@@ -44,132 +32,91 @@ const fetchUser: UserFetcherSignature = async (userId, isLoadingAuth) => {
...
-};
-
-const deleteDevice = async (
-  id: DevicesType["id"],
-  userId: string | undefined
+const createSensor = async (
+  sensor: Omit<definitions["sensors"], "id">
 ): Promise<void> => {
-  if (!userId) throw new Error("Not authenticated");
+  if (!sensor.user_id) throw new Error("Not authenticated");
```
Comment (vogelino) [inline src/lib/hooks/useUserData/index.ts:61]: Here it isn't about the sensor's `user_id` but it's about verifying that the user is authenticated. This should come from `useAuth` or similar. It represents the id of the logged-in user.

Comment (vogelino) [reply src/lib/hooks/useUserData/index.ts:61]: My bad. In the diff, this looks as if you're deleting the sensor but you are creating one. Ignore this

File: src/lib/hooks/useUserData/index.ts [hunk 2]
```diff
@@ -44,132 +32,91 @@ const fetchUser: UserFetcherSignature = async (userId, isLoadingAuth) => {
...
 
-const deleteProject = async (
-  id: number,
-  userId: string | undefined
+const deleteSensor = async (
+  id: definitions["sensors"]["id"],
+  user_id: string | undefined
 ): Promise<void> => {
-  if (!userId) throw new Error("Not authenticated");
+  if (!user_id) throw new Error("Not authenticated");
```
Comment (vogelino) [inline src/lib/hooks/useUserData/index.ts:113]: Yeah see, in this function you're doing it right! 👌

File: src/lib/hooks/useUserData/index.ts [hunk 3]
```diff
@@ -181,14 +128,15 @@ const updateUser = async (
...
+    .from<definitions["user_profiles"]>("user_profiles")
     .update({ name: newName })
     .eq("id", userId);
 
   if (nameReset.error) throw nameReset.error;
 };
 
-const updateEmail = async (
+// We currently don't support updating the email. TODO: potentially delete this function
+/* const updateEmail = async (
```
Comment (vogelino) [inline src/lib/hooks/useUserData/index.ts:139]: We should delete these lines and not leave commented code in there.

File: src/lib/hooks/useUserData/index.ts [hunk 4]
```diff
@@ -212,118 +160,79 @@ const deleteUser = async (userId: string | undefined): Promise<void> => {
...
-    project: Omit<ProjectsType, "id" | "createdAt">
-  ) => Promise<ProjectsType[] | null | void>;
-  updateProject: (project: Omit<ProjectsType, "createdAt">) => Promise<void>;
-  deleteProject: (id: number) => Promise<void>;
+  createSensor: (sensor: Omit<definitions["sensors"], "id">) => Promise<void>;
+  updateSensor: (sensor: Partial<definitions["sensors"]>) => Promise<void>;
+  deleteSensor: (id: number) => Promise<void>;
   updateUser: (newName: string) => Promise<void>;
-  updateEmail: (newEmail: string) => Promise<void>;
+  /* updateEmail: (newEmail: string) => Promise<void>; */
```
Comment (vogelino) [inline src/lib/hooks/useUserData/index.ts:170]: Same here.

File: src/lib/hooks/useUserData/index.ts [hunk 5]
```diff
@@ -212,118 +160,79 @@ const deleteUser = async (userId: string | undefined): Promise<void> => {
...
+      void mutate(sensorsParams);
     },
     updateUser: async name => {
       if (user.data?.name === name) return;
       void mutate(userParams, { ...user, name }, false);
       await updateUser(name, userId).catch(setActionError);
       void mutate(userParams);
     },
-    updateEmail: async email => {
+    /* updateEmail: async email => {
```
Comment (vogelino) [inline src/lib/hooks/useUserData/index.ts:230]: Same here.

File: src/lib/hooks/useUserData/manageSensorsLocally.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { definitions } from "@common/types/supabase";
+
+let lastSensorId = 10000000;
+
+export const updateSensorsLocally = (
```
Comment (vogelino) [inline src/lib/hooks/useUserData/manageSensorsLocally.ts:5]: Great and clean implementation here. Very well done 💪 !

File: src/lib/requests/getSensorData/getSensorData.test.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    } catch (error) {
+      expect(error).toEqual({
+        message: "Error message",
+      });
+    }
+    server.resetHandlers();
+    server.close();
+  });
+
+  /* it.skip("should send error message when erroring", async (): Promise<void> => {
```
Comment (vogelino) [inline src/lib/requests/getSensorData/getSensorData.test.ts:48]: We should delete these comments

File: src/lib/requests/getSensorLastRecordDate/index.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    .from<Pick<definitions["records"], "id" | "recorded_at" | "sensor_id">>(
+      "records"
+    )
+    .select("id,recorded_at")
+    .eq("sensor_id", sensorId)
+    .order("recorded_at", { ascending: false })
+    .limit(1);
+
+  if (error) throw error;
+  if (!records) throw new Error(`No record found for device ID ${sensorId}`);
```
Comment (vogelino) [inline src/lib/requests/getSensorLastRecordDate/index.ts:17]: device -> sensor

File: src/mocks/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# This folder contains all the mocking data generated with Mock Service Worker (MSW)
+
+**Attention**: Mocking with MSW may not work properly in Firefox. If you run into issues, first try to switch to Chrome for the time being. We are aware that there is an issue, but consider fixing this low prio.
```
Comment (vogelino) [inline src/mocks/README.md:3]: Good advice. Thx for adding note in the README

**Review by vogelino — CHANGES_REQUESTED** [review]
Great work! Especially due to the spread of the changes! Very well done! I found a few minimal spots where you forgot to change a term, comment or so. I guess we should search one last time for the terms "Project", "Projekt", "Gerät", "Device", etc. to make sure we didn't forget anything.

File: src/__tests__/api/v1/devices/[id]/records.test.ts [hunk 10]
```diff
@@ -4,8 +4,10 @@ import { createApiUrl } from "../../../../../lib/requests/createApiUrl";
 import { createMocks } from "node-mocks-http";
 import recordsHandler from "../../../../../../pages/api/v1/devices/[id]/records";
 import { VALID_TIMESTAMP_EXAMPLE } from "@lib/timestampValidator";
-import { fakeDeviceWithFewRecords as fakeDevice } from "@mocks/supabaseData/publicProjects";
 import { NextApiRequest, NextApiResponse } from "next";
+import { sensors } from "@mocks/supabaseData/sensors";
```
[the remark this answers is not in the document]
Comment (dnsos) [reply src/__tests__/api/v1/devices/[id]/records.test.ts:8]: True, I changed that in c33ca26. As @ff6347 is starting to build these routes in the Fastify API, we will have to delete the frontend API routes soon-ish anyway.

File: pages/sitemap.xml.ts [hunk 11]
```diff
@@ -13,15 +12,15 @@ const formatDate: (dateStr?: string) => string = dateStr => {
...
 <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
   <url>
     <loc>${createFullUrl("/")}</loc>
     <lastmod>${formatDate()}</lastmod>
   </url>
-  ${projects
+  ${sensors
     .map(
       ({ id }) => `
   <url>
```
[the remark this answers is not in the document]
Comment (dnsos) [reply pages/sitemap.xml.ts:26]: Done in 4f8b696.

File: src/lib/hooks/useUserData/index.ts (hunk 1 above)
Comment (dnsos) [reply src/lib/hooks/useUserData/index.ts]: You're actually right, we should get the `user_id` from `useAuth`. I changed that now in 6f72974.

**Comment by dnsos** [comment]
> I guess we should search one last time for the terms "Project", "Projekt", "Gerät", "Device", etc. to make sure we didn't forget anything.

Yes. There are a few leftovers:

- The documentation still talks a lot about devices and projects. This is a separate ticket and not handled here (@julizet)
- A few components are still named after the old project concept. For now I have marked them as `_DEPRECATED`. I would suggest we keep them in the repo as inspiration for building out the rest of the new UI because it is easier than searching through the Git history for old files. What do you think?

**Review by vogelino — APPROVED** [review]
Perfect! Thank you! :)

**Pull request merged**